apps/total-typescript/src/components/app/navigation.tsx

```diff
@@ -70,38 +68,175 @@ const DesktopNav = () => {
           <FireIcon className="h-5 w-5 text-orange-400" aria-hidden="true" />
         }
       />
+      {isSellingLive && (
+        <hr
+          className="mx-3 h-1/4 w-px border-transparent bg-gray-700"
+          aria-hidden="true"
+        />
+      )}
+      {canViewContent ? (
+        <AccountDropdown />
+      ) : (
+        isSellingLive && <NavLink path="/login" label="Log in" />
+      )}
     </ul>
   )
 }
 
+const MobileNav = () => {
+  const ability = useAbilities()
+  const canViewTeam = ability.can('view', 'Team')
+  const canViewContent = ability.can('view', 'Content')
+  const canViewInvoice = ability.can('view', 'Invoice')
+
+  return (
+    <div className="block md:hidden">
+      <ul className="flex h-full items-center justify-center">
+        {isSellingLive && !canViewContent && (
+          <NavLink path="/login" label="Log in" />
+        )}
+        <li className="h-full">
+          <NavigationMenu.Root delayDuration={0} className="flex h-full">
+            <NavigationMenu.List className="flex h-full items-center justify-center">
+              <NavigationMenu.Item className="h-full">
+                <NavigationMenu.Trigger className="-mr-3 flex h-full items-center justify-center px-5 hover:bg-gray-800 sm:-mr-5">
+                  <MenuIcon className="h-4 w-4" aria-hidden />
+                  <span className="sr-only">Menu</span>
+                </NavigationMenu.Trigger>
+                <NavigationMenu.Content className="absolute left-0 top-full w-full bg-gray-800 shadow-xl">
+                  <ul>
+                    <MobileNavLink
+                      path="/tutorials"
+                      label="Free Tutorials"
+                      icon={
+                        <PlayIcon
+                          className="h-5 w-5 text-cyan-300"
+                          aria-hidden="true"
+                        />
+                      }
+                    />
+                    <MobileNavLink
+                      path="/tips"
+                      label="Tips"
+                      icon={
+                        <FireIcon
+                          className="h-5 w-5 text-orange-400"
+                          aria-hidden="true"
+                        />
+                      }
+                    />
+                    {canViewContent && isSellingLive && (
```

Comment:
I don't know if this check will work for this app. I believe a user who has team ownership (ability to manage a team) may not necessarily be able to view content (they haven't redeemed a purchase for themselves). Plus, the ability checks are typically done on a per module/section/lesson basis. The navigation often won't have the context -- for example, if the user is on the home page.

Thinking about what an alternative could be 🤔

Reply:
What if we add an ability like `can('manage', 'Account')` for when we have a signed in user. If they can manage their account, then they can see the drop down which will allow them to click 'Sign Out'.

![CleanShot 2022-11-22 at 08 55 56@2x](https://user-images.githubusercontent.com/694063/203346207-b83a92c4-39ef-4c76-b562-913f14c22533.png)

Reply:
I could also see `can('view', 'Account')` -- not sure which one makes more sense.

Reply:
ability checks can be done anywhere at any time on either client or server, so there's no "typically" just where it was done initially to protect the content. Using `manage` should probably be avoided since it's a "yes" to all actions, in favor of explicit actions.

I'd probably show this is `user` exists, and then be fine grained about what actions are present beyond that and then have `login` if user is not present

Reply:
> ability checks can be done anywhere at any time on either client or server, so there's no "typically"

What I meant by that was, the `can('view', 'Content')` only makes sense in the context of a specific module/section/lesson. Since the navigation is requesting ability checks without any content specified, the `view content` check will never resolve true.